### PR TITLE
completes the app functionality

### PR DESCRIPTION
**Pull request by AdoshSingh**
Completes the app functionality integrating frontend and backend.
Setup for using supabase locally.
Added seeding script for seeding database.
Added CI pipeline for the app.
Updated readme.

### CODE REVIEW

**Review by dray92 — COMMENTED** [review]
overall, lgtm

File: roadmap-voting-app/app/api/user/route.ts [hunk 1]
```diff
@@ -4,15 +4,28 @@ import { supabase } from '../../../lib/supabaseClient';
...
     try {
         const login = req.nextUrl.searchParams.get("login") as string;
-        const formdata = await req.formData();
-        const email = formdata.get("email") as string;
-        const password = formdata.get("password") as string;
         if (login) {
-            const { data , error } = await supabase.auth.signInWithPassword({ email, password });
-            if (error) throw error;
-            return NextResponse.json({ message: "User logged in successfully", data: data, status: 200 });
+            const formdata = await req.formData();
```
[the remark this answers is not in the document]
Comment (AdoshSingh) [reply roadmap-voting-app/app/api/user/route.ts]: My bad, I think it got changed when I tried to bring the changes in the main branch to this branch. Took care of it now.

File: roadmap-voting-app/app/api/user/route.ts [hunk 2]
```diff
@@ -4,15 +4,28 @@ import { supabase } from '../../../lib/supabaseClient';
...
+                .from('userstable')
+                .select('*')
+                .eq('id', userId)
+                .single();
+
+            if (userError) throw userError;
+            return NextResponse.json({ message: "User logged in successfully", data: signInresponse.data, userData: userData, status: 200 });
         } else {
-            const username = formdata.get("username") as string;
+            const formdata = await req.formData();
```
[the remark this answers is not in the document]
Comment (AdoshSingh) [reply roadmap-voting-app/app/api/user/route.ts]: My bad, I think it got changed when I tried to bring the changes in the main branch to this branch. Took care of it now.

File: roadmap-voting-app/components/Name.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        loadName();
+        window.addEventListener("storageUpdate", loadName);
+
+        // Clean up the event listener
+        return () => {
+            window.removeEventListener("storageUpdate", loadName);
+        };
+    }, []);
+
+    // useEffect(() => {
```
[the remark this answers is not in the document]
Comment (AdoshSingh) [reply roadmap-voting-app/components/Name.tsx]: Yes, I have removed this unnecessary comment.

File: roadmap-voting-app/components/Post.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    const vote = (type: string): void => {
+        const localData = localStorage.getItem("session");
+        const session = localData ? JSON.parse(localData) : null;
+        if (!session) {
+            alert('Please login to vote');
+            return;
+        }
+        const userID = session?.user?.id;
+        const token = session?.access_token;
+        console.log(token);
```
[the remark this answers is not in the document]
Comment (AdoshSingh) [reply roadmap-voting-app/components/Post.tsx]: Also removed this, It was logged for some debugging.

File: roadmap-voting-app/lib/authenticate.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import {supabase} from './supabaseClient';
+import { NextRequest } from 'next/server';
+
+export async function authenticate(req: NextRequest) {
+    const token = req.headers.get('Authorization')?.split(' ')[1];
+    // spelling dekhio
```
[the remark this answers is not in the document]
Comment (AdoshSingh) [reply roadmap-voting-app/lib/authenticate.ts]: Sorry this was a comment left for me by a fellow developer who was working with me in this and forgot to delete this comment. Took care of it as well.

File: roadmap-voting-app/app/page.tsx [hunk 6]
```diff
@@ -1,7 +1,50 @@
...
 export default function Home() {
+
+  const [posts, setPosts] = useState<post[]>();
+
+  useEffect(() => {
+
+    fetch("/api/post")
+      .then((res) => res.json())
+      .then((data) => {
+        // console.log(data);
```
Comment (dray92) [inline roadmap-voting-app/app/page.tsx:26]: remove

File: roadmap-voting-app/components/Post.tsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        if (target) {
+            getDate(target);
+        }
+    }, [target])
+
+    useEffect(() => {
+        fetch(`/api/post?postId=${id}`)
+            .then((res) => res.json())
+            .then((data) => {
+                console.log(data);
```
Comment (dray92) [inline roadmap-voting-app/components/Post.tsx:51]: remove?

File: roadmap-voting-app/components/Post.tsx [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            fetch(`/api/vote`, {
+                method: 'POST',
+                headers: {
+                    "Authorization": `Bearer ${token}`,
+                },
+                body: formData,
+            })
+                .then((res) => res.json())
+                .then((data) => {
+                    console.log(data);
```
Comment (dray92) [inline roadmap-voting-app/components/Post.tsx:98]: remove?

**Review by dray92 — APPROVED** [review]

**Comment by AdoshSingh** [comment]
I have removed any unecessary console.logs and comments from the code

File: roadmap-voting-app/app/page.tsx (hunk 6 above)
Comment (AdoshSingh) [reply roadmap-voting-app/app/page.tsx]: removed

File: roadmap-voting-app/components/Post.tsx (hunk 7 above)
Comment (AdoshSingh) [reply roadmap-voting-app/components/Post.tsx]: removed

File: roadmap-voting-app/components/Post.tsx (hunk 8 above)
Comment (AdoshSingh) [reply roadmap-voting-app/components/Post.tsx]: removed

**Pull request merged**